packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1417,6 +1417,57 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
     return child;
   }
 
+  /// The delegated transition provided to the previous route.
+  DelegatedTransitionBuilder? get delegatedTransition => null;
+
+  /// The delegated transition received from an incoming route.
+  DelegatedTransitionBuilder? receivedTransition;
+
+  /// Transition that matches with the next route. Will be null if there is no
+  /// next route or a next route is not provided. If a non-null value exists,
+  /// then it will be used in place of [buildTransitions].
+  DelegatedTransitionBuilder? get nextRouteTransition {
+    if (receivedTransition != null) {
+      return receivedTransition;
+    } else {
+      return null;
+    }
+  }
+
+  /// Wraps the transitions of the route with a delegated transition received
```

Comment:
"the route" => "this route"

"delegatedTransition" => "[delegatedTransition]" (or something we can link to with more info)